### ISSUE TITLE
[Feature] enable currency number formatting

### ISSUE THREAD

**Issue by areyabhishek** [appsmithorg/appsmith#6942]
## Summary
Numbers need to be shown in formats: 10,00,000 or 1,000,000 or 1,000,000.00, etc. 
We should show a condition that shows these options in the property pane.

**Comment by somangshu** [comment]
@areyabhishek are you talking about the table widget or the input widget.

Also, would something like a toggle called as `internationalised format` work?

@Nikhil-Nandagopal can you think of a better terminology here?

**Comment by Nikhil-Nandagopal** [comment]
I think just the number format is good. This needs to be a dropdown where a user can select the format (Check Google Sheets format option). This should only be in places where the data type is known i.e Input & Table. Users should use a JS lib for places like text

**Comment by somangshu** [comment]
@Nikhil-Nandagopal I didn't quiet get it! 
- We want this only on the number / currency type in the input  / table widget prop pane?
- If this a dropdown, what are the options, can you please suggest?
- Do you think this is only valid for the currency type?

**Comment by Nikhil-Nandagopal** [comment]
It's valid for any numerical type including currently. See the screenshots 
<img width="273" alt="Screenshot 2021-09-08 at 4 14 35 PM" src="https://user-images.githubusercontent.com/3897254/132495939-a7ba4d91-3159-43bc-b057-f5775e65ece6.png">
from google sheets
<img width="522" alt="Screenshot 2021-09-08 at 4 14 18 PM" src="https://user-images.githubusercontent.com/3897254/132495925-0fb5db64-7376-4ba6-8439-b7f8ed9ae277.png">

**Comment by somangshu** [comment]
Oops, I was checking the appsmith google integration 🙈 

Got it now, thanks.

**Comment by vicky-primathon** [comment]
@somangshu  needs design for this

**Comment by vicky-primathon** [comment]
- [ ] Add a number formats dropdown in Input Widget property pane
- [ ] Add a number formats dropdown in Table Widget property pane
- [ ] Selected number format should be applied for for number type columns in Table Widget
- [ ] Selected number format should be applied for for number type Input widget
- [ ] Add cypress test to validate number formatting in Table widget
- [ ] Add cypress test to validate number formatting in Input widget

**Comment by dilippitchika** [comment]
We can follow something similar to the property exposed in date fields, which is display date format. 

Reach - Users configuring the table's col as number type
Impact - Low 
Effort - Assuming 1 sprint for dev+design+QA

## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 500 |
| Effort (months)  | 0.5 |

**Comment by somangshu** [comment]
@dilippitchika I think this should also be a part of the input widget! Under the type number these formatting options are valid for the user

**Comment by dilippitchika** [comment]
@somangshu i was only considering this for display purpose and not the editing purpose yet. That is the reason i removed the input widget. 

If we were to add it there as well, i believe the issue should be updated, because the impact is for app-viewer will be higher because they can edit the values.

**Comment by somangshu** [comment]
@dilippitchika in that case, I also want to think about how this will pan out for table widget inline edit, Will be persist the formating or remove it before the editing starts. 

another point is that this can we the output also, like {{input1.text}} gives out the formatted label required many times by the DB

**Comment by Nikhil-Nandagopal** [comment]
I think it should work exactly like google sheets or our date picker. The underlying format is a number and you input it as a number in edit mode but there is presentation logic to add commas

**Comment by bobbintb** [comment]
I came across a need for this recently. I have file sizes in a column that are difficult to read in the default format (bytes). I have a JS function to make it more human readable but that makes it impossible to sort. I even tried changing it to a date type and using the custom JS but that did not work. I think something similar to the date type would work well, and give formatting option like Google Sheets.

**Comment by Tooluloope** [comment]
@bobbintb are you formatting the column after it's been rendered on the table, or are you formatting the data before it gets to the table?

**Comment by bobbintb** [comment]
> @bobbintb are you formatting the column after it's been rendered on the table, or are you formatting the data before it gets to the table?

I'm formatting before it gets to the table via the computed value in the widget. I'm not sure how I would do it otherwise.

**Comment by Tooluloope** [comment]
You can format the data just before it gets to the table, not the column.

**Comment by bobbintb** [comment]
> You can format the data just before it gets to the table, not the column.

I'm not quite following. The table widget displays data from a MongoDB query. If I leave the data I can sort it but it's difficult to read. If I format it I can read it better (ex 93.05 MB instead of 97569381) but it breaks sorting because it is no longer a number. Are you saying I can format it and keep sorting functionality or would this require a new feature?

**Comment by bobbintb** [comment]
It doesn't unfortunately. It still transforms the data. The sort works with the specific test numbers you've chosen but I think it will break like mine did with more varied examples (KB, bytes, GB, etc).

**Comment by Tooluloope** [comment]
Oh got it. This won't work with varying bytes unit

**Comment by Tooluloope** [comment]
@dilippitchika @sbalaji1192, is there another way to unblock him?

**Comment by sbalaji1192** [comment]
@Tooluloope @dilippitchika I don't think it is possible without a property to format numbers while rendering the cells like the date output format property.
